### PR TITLE
ACCSEC-21631 Unit tests for AndroidKeyStore

### PR DESCRIPTION
**Pull request by yafuquen**
## Jira Ticket
- ACCSEC-21631

## Description
- Add tests for new encrypt and decrypt methods in AndroidKeyStore. 
- Add test for EncryptedPreferences using commit

## Testing
- [x] Added unit tests
- [x] Ran unit tests successfully
- [ ] Added documentation for public APIs and/or Wiki

### CODE REVIEW

**Review by AlejandroOrozco — APPROVED** [review]
LGTM 👍

File: security/src/test/java/com/twilio/security/storage/EncryptedPreferencesTest.kt [hunk 1]
```diff
@@ -74,6 +74,25 @@ class EncryptedPreferencesTest {
...
+    val value = "value"
+    val editor: Editor = mock()
+    whenever(preferences.edit()).thenReturn(editor)
+    whenever(serializer.toByteArray(value)).thenReturn(value.toByteArray())
+    whenever(secretKeyProvider.encrypt(any())).thenReturn(value.toByteArray())
+    whenever(editor.putString(eq(generateKeyDigest(key)), any())).thenReturn(editor)
+    whenever(editor.commit()).thenReturn(false)
+    exceptionRule.expect(StorageException::class.java)
+    exceptionRule.expectCause(
+        instanceOf(
```
Comment (AuthyApps) [inline security/src/test/java/com/twilio/security/storage/EncryptedPreferencesTest.kt:89]: <table data-meta="generated_by_Linter">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-android/blob/5b0eb61d7d2ddb2c184e8c32f649596ce888d592/security/src/test/java/com/twilio/security/storage/EncryptedPreferencesTest.kt#L89"></span>Unexpected indentation (8) (should be 6)</td>
    </tr>
  </tbody>
</table>

File: security/src/test/java/com/twilio/security/storage/EncryptedPreferencesTest.kt [hunk 2]
```diff
@@ -74,6 +74,25 @@ class EncryptedPreferencesTest {
...
+            IllegalStateException::class.java
```
Comment (AuthyApps) [inline security/src/test/java/com/twilio/security/storage/EncryptedPreferencesTest.kt:90]: <table data-meta="generated_by_Linter">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-android/blob/5b0eb61d7d2ddb2c184e8c32f649596ce888d592/security/src/test/java/com/twilio/security/storage/EncryptedPreferencesTest.kt#L90"></span>Unexpected indentation (12) (should be 8)</td>
    </tr>
  </tbody>
</table>

File: security/src/test/java/com/twilio/security/storage/EncryptedPreferencesTest.kt [hunk 3]
```diff
@@ -74,6 +74,25 @@ class EncryptedPreferencesTest {
...
+        )
```
Comment (AuthyApps) [inline security/src/test/java/com/twilio/security/storage/EncryptedPreferencesTest.kt:91]: <table data-meta="generated_by_Linter">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-android/blob/5b0eb61d7d2ddb2c184e8c32f649596ce888d592/security/src/test/java/com/twilio/security/storage/EncryptedPreferencesTest.kt#L91"></span>Unexpected indentation (8) (should be 6)</td>
    </tr>
  </tbody>
</table>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Fix code convention problems</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request title does not follow the convention [ACCSEC-0000] PR Title text</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Fix code convention problems</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request title does not follow the convention [ACCSEC-0000] PR Title text</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request title does not follow the convention [ACCSEC-0000] PR Title text</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by SergioFierro — APPROVED** [review]

**Pull request merged**